### PR TITLE
DeveloperGuide: Fix diagrams, typos and add EditCommand diagram

### PR DESCRIPTION
**Pull request by EmilyOng**
# Diagrams Changed

## Delete Command Sequence Diagram

- Update the argument string to contain `1 t/e` in accordance with PennyWise's format
- Add more details in the interaction between `DeleteCommand` and `Model`
- **Follow-up**: It would be better to shift this interaction into a separate reference frame.

<details>
<summary>Diagram</summary>

| Before | After |
| ------- | ------- |
| image | image |

</details>

## Edit Command Sequence Diagram

- **Question**: Can we display private methods in the sequence diagram? I included it, such as `getEntryToEdit`.
- **Follow-up**: It would be better to shift this interaction into a separate reference frame.

<details>
<summary>Diagram</summary>

image

</details>

## Pie Chart View Sequence Diagram

I removed the dangling cross that you see below:

<details>
<summary>Diagram</summary>

image

</details>

## View Activity Diagram

- **Question**: I was wondering if this level of abstraction is enough for the activity diagram? Should it be more specific? I kept it at this level since the lower-level details are covered in the sequence diagram

<details>
<summary>Diagram</summary>

image

</details>

# Glossary

- Added the `Entry` definition

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 1]
```diff
@@ -57,10 +57,8 @@ public CommandResult execute(Model model) throws CommandException {
             if (targetIndex.getZeroBased() >= lastShownList.size()) {
                 throw new CommandException(Messages.MESSAGE_INVALID_ENTRY_DISPLAYED_INDEX);
             }
-            entryToDelete = lastShownList.get(targetIndex.getZeroBased());
```
Comment (EmilyOng) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:60]: Not used

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 2]
```diff
@@ -63,6 +63,14 @@ public EditCommand(Index index, EditEntryDescriptor editEntryDescriptor) {
         this.entryType = editEntryDescriptor.getType().get();
     }
 
+    private Entry getEntryToEdit(Index targetIndex, List<Entry> lastShownList)
```
Comment (EmilyOng) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:66]: Added this method so it's easier to draw in the sequence diagram, but can we add private methods to the sequence diagram?

File: docs/diagrams/EditSequenceDiagram.puml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+!include style.puml
+
+box Logic LOGIC_COLOR_T1
+participant ":LogicManager" as LogicManager LOGIC_COLOR
+participant ":PennyWiseParser" as PennyWiseParser LOGIC_COLOR
+participant ":EditCommandParser" as EditCommandParser LOGIC_COLOR
+participant "editCommand:EditCommand" as EditCommand LOGIC_COLOR
+participant "editEntryDescriptor:EditEntryDescriptor" as EditEntryDescriptor LOGIC_COLOR
+participant ":CommandResult" as CommandResult LOGIC_COLOR
+participant "<<class>>\nEditCommand" as EditCommandClass LOGIC_COLOR
```
Comment (EmilyOng) [inline docs/diagrams/EditSequenceDiagram.puml:11]: For the class method `createdEditedEntry`

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -270,6 +268,39 @@ Step 10. The `AddCommand` then creates a `CommandResult` instance and returns it
   * Pros: Users can be more flexible in grouping their spending/incomes.
   * Cons: Possible dilution of categories, which would make the PieChart diagram not as useful.
 
+### Edit Entry
```
Comment (EmilyOng) [inline docs/DeveloperGuide.md:271]: As I was writing this, it seemed like we are repeating what is drawn in the sequence diagram which may be a DG bug. Do you think we should clean these up in another PR by just summarizing the steps?

File: src/main/java/seedu/address/logic/commands/EditCommand.java (hunk 2 above)
Comment (JasonYapzx) [reply src/main/java/seedu/address/logic/commands/EditCommand.java:66]: Im unable to find private methods in the sequence diagrams in our textbook, I searched online as well and they mentioned it is okay to put if we want the developers to understand our implementation, but it is also okay not to put.

What do the rest think?

@AY2223S1-CS2103T-W17-2/developers

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (JasonYapzx) [reply docs/DeveloperGuide.md:271]: If this counts as repetitive details then I think our reviewers might see it as a DG bug, I am okay to summarise the steps in another PR. What does the rest think

@AY2223S1-CS2103T-W17-2/developers

File: docs/diagrams/ViewActivityDiagram.puml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      :Retrieve the list of income;
+    case ([entry type is expenditure])
+      :Retrieve the list of expenditure;
+    case ([else])
+      :Display an error message;
+      stop
+    endswitch
+
+    switch ()
+    case ([category graph type is specified])
```
Comment (yuanxi1) [inline docs/diagrams/ViewActivityDiagram.puml:17]: Are there checks for presence of `mo/` param when graph type is `month`. Should we add this in the activity diagram?

**Review by yuanxi1 — APPROVED** [review]

File: docs/diagrams/ViewActivityDiagram.puml (hunk 5 above)
Comment (yuanxi1) [reply docs/diagrams/ViewActivityDiagram.puml:17]: I think the level of abstraction is quite good for the activity diagram.

File: docs/diagrams/ViewActivityDiagram.puml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+      :Display the list of entries;
+      :Display pie chart of the entries\nbased on all categories;
+    case ([month graph type is specified])
+      :Display the list of entries in the month;
+      :Display line chart of amount statistic\nof the entries in the month;
+    endswitch
+else ([end])
```
Comment (yuanxi1) [inline docs/diagrams/ViewActivityDiagram.puml:24]: Should this be `[else]` instead of `[end]` since the if condition is `[inputs are valid]`

**Pull request merged**